EIPS/eip-2294.md

```diff
@@ -0,0 +1,40 @@
+---
+eip: 2294
+title: Explicit bound to Chain ID size
+description: Adds a maximum value to the Chain ID parameter to avoid potential encoding issues that may occur when using large values of the parameter.
+author: Bryant Eisenbach (@fubuloubu), Alex Beregszaszi (@axic), Zainan Victor Zhou (@xinbenlv)
+discussions-to: https://ethereum-magicians.org/t/eip-2294-explicit-bound-to-chain-id/11090
+status: Draft
+type: Standards Track
+category: Core
+created: 2019-09-19
+requires: 155
+---
+
+## Abstract
+This EIP restricts the size of [EIP-155](./eip-155.md) Chain ID parameter to a particular maximum value `floor(MAX_UINT64 / 2) - 36`, in order to ensure that there is a standard around how this parameter is to be used between different projects.
+
+## Motivation
+EIP-155 introduces the Chain ID parameter, which is an important parameter used for domain separation (replay protection) of Ethereum protocol signed messages. However, it does not specify any properties about the size that this parameter takes. @axic mentions this concern in [EIP-1344 (comment)](https://ethereum-magicians.org/t/eip-1344-add-chain-id-opcode/1131/109) and suggests a reasonable maximum enforced size in order to ensure that there are no issues when encoding this parameter. This would allow a sufficient amount of different values for this parameter, which is typically chosen by community consensus as a genesis parameter for a given chain and thus does not change often.
```

Comment:
Requesting a manual approval from editors for including this link
```
https://ethereum-magicians.org/t/eip-1344-add-chain-id-opcode/1131/109
```
@axic @SamWilsn @lightclient @gcolvin @Pandapip1

Reply:
I think you could probably do something like:

```markdown
@axic mentions this concern in [EIP-1344](./eip-1344.md)'s discussion thread.
```

Reply:
@SamWilsn this is the url, it is not exactly the eip-1344's discussion thread.

https://ethereum-magicians.org/t/eip-1344-add-chain-id-opcode/1131/109
Pin to a particular response (109) of that thread which is necessary for the supporting argument in this content. 

What's the justification for disallowing the very discussion link we (EIP repo) _required_ for all EIPs?

Reply:
> What's the justification for disallowing the very discussion link we (EIP repo) required for all EIPs?

Any important considerations would eventually be included in the EIP.

Reply:
Fine. I will remove the links for now until we have a clearer external link policy.

For the record, I respectively protest your interpretation of our link policy, my friend @Pandapip1 .